Sources/GRPCHTTP2Core/Server/HTTP2ServerTransport.swift

```diff
@@ -188,4 +188,162 @@ extension HTTP2ServerTransport.Config {
       Self(maxRequestPayloadSize: 4 * 1024 * 1024)
     }
   }
+
+  public struct TransportSecurity: Sendable {
+    private enum Wrapped {
+      case plaintext
+      case tls(TLS)
+    }
+
+    private let wrapped: Wrapped
+
+    /// This connection is plaintext: no encryption will take place.
+    public static let plaintext = Self(wrapped: .plaintext)
+
+    /// This connection will use TLS.
+    public static func tls(_ tls: TLS) -> Self {
+      Self(wrapped: .tls(tls))
+    }
+
+    /// Returns the TLS configuration, if the security has been set to ``tls(_:)``.
+    public var tlsConfig: TLS? {
+      switch wrapped {
+      case .plaintext:
+        return nil
+      case .tls(let config):
+        return config
+      }
+    }
+  }
+
+  public struct TLS: Sendable {
+    /// The serialization format of the provided certificates and private keys.
+    public struct SerializationFormat: Sendable, Equatable {
+      private enum Wrapped {
+        case pem
+        case der
+      }
+
+      private let serialization: Wrapped
+
+      public static let pem = Self(serialization: .pem)
+      public static let der = Self(serialization: .der)
+    }
+
+    public struct CertificateSource: Sendable {
+      private enum Wrapped {
+        case file(path: String, serializationFormat: SerializationFormat)
+        case certificate(bytes: [UInt8], serializationFormat: SerializationFormat)
+      }
+
+      private let wrapped: Wrapped
+
+      /// The certificate will be provided via a file.
+      public static func file(path: String, serializationFormat: SerializationFormat) -> Self {
+        Self(wrapped: .file(path: path, serializationFormat: serializationFormat))
+      }
+
+      /// The certificate will be provided as an array of bytes.
+      public static func certificate(
```

Comment:
I think this source should be called `bytes`. It doesn't make much sense for the source of a certificate to be a certificate.